uspace/app/bdsh/cmds/modules/ls/ls.c

```diff
@@ -89,14 +97,45 @@ static unsigned int ls_start(ls_job_t *ls)
  */
 static void ls_print(struct dir_elem_t *de)
 {
-	if (de->s.is_file)
-		printf("%-40s\t%llu\n", de->name, (long long) de->s.size);
-	else if (de->s.is_directory)
-		printf("%-40s\t<dir>\n", de->name);
+	int width = 13;
+
+	if (de->s.is_file) {
+		if (ls.well_formatted) {
+			cap_spec_t cap;
+
+			cap_from_blocks(1, de->s.size, &cap);
```

Comment:
If anything, this should be cap_from_blocks(de->s.size, 1, &cap) - the file size can be 64-bit, while size_t can only hold 32-bits on 32-bit CPU.